arch/Kconfig

```diff
@@ -962,6 +962,20 @@ config EXTERNAL_CACHE
 
 endchoice
 
+config DCACHE_AUTO_ENABLE
```

Comment:
Why do we need new configs?
Aren't DCACHE and ICACHE enough?

In the others commits of this PR, I see you changed the `IS_ENABLED(CONFIG_{I,D}CACHE)` with `IS_ENABLED({I,D}CACHE_AUTO_ENABLE)`

Reply:
Because I think it might make sense that you want to have the caches available, but not automatically enabled. Hence, both options.

Reply:
Basically, I've generalized the approach of CONFIG_IMXRT1XXX_CODE/DATA_CACHE and CONFIG_KINETIS_KV5X_ENABLE_CODE/DATA_CACHE.

Reply:
> Because I think it might make sense that you want to have the caches available, but not automatically enabled. Hence, both options.

Sorry, but I still don't see a need for this.
Right now, there is no case where you have DCACHE/ICACHE but you don't want them automatically enabled.
If there was a use case as you mentioned, maybe it would make sense, but now, there is no such case.

Reply:
> Basically, I've generalized the approach of CONFIG_IMXRT1XXX_CODE/DATA_CACHE and CONFIG_KINETIS_KV5X_ENABLE_CODE/DATA_CACHE.

I think these configs `CONFIG_IMXRT1XXX_CODE/DATA_CACHE` and `CONFIG_KINETIS_KV5X_ENABLE_CODE/DATA_CACHE` should be changed to `ICACHE/DCACHE`.
But, please @danieldegrasse maybe you know better why these were added.